pkg/types/openstack/machinepool.go

```diff
@@ -17,4 +20,21 @@ func (o *MachinePool) Set(required *MachinePool) {
 	if required.FlavorName != "" {
 		o.FlavorName = required.FlavorName
 	}
+
+	if required.RootVolume != nil {
+		if required.RootVolume.Size != 0 {
+			o.RootVolume.Size = required.RootVolume.Size
+		}
+		if required.RootVolume.Type != "" {
```

Comment:
If valid `MachinePool` set both `Size` and `Type`, it is surprising to me that we have two separate conditionals here.  Can we validate `required` and then just use:

```golang
if required.RootVolume.Size != 0 {
  o.RootVolume.Size = required.RootVolume.Size
  o.RootVolume.Type = required.RootVolume.Type
}
```

Do we actually use this `Set` call?  If so, maybe `type` and `size` actually _are_ optional as long as someone in this chain defines them?